File: 연습문제/LV.1/정수 내림차순으로 배치하기.py
#LV.1 정수 내림차순으로 배치하기.py
#주어진 int를 뒤짚어 출력하는 알고리즘
#
#문자열변환 후, sorted 함수 이용 => join 및 int함수로 변환 
#

def solution(n):
    
    answer = []
    for i in str(n) :
        answer.append(i)
    #answer = list(str(n))     
   
        
    #print(answer)
    #print(sorted(answer, key= lambda x : x[0], reverse=True))
    
    answered = sorted(answer, key= lambda x : x[0], reverse=True)
    
    return int(''.join(answered))
    
    

File: 연습문제/LV.1/정수 내림차순으로 배치하기.java
import java.util.*;


class Solution {
    public long solution(long n) {
        long answer = 0;
        
        
        String[] str = String.valueOf(n).split("");
        //System.out.println(Arrays.toString(str));       

        Arrays.sort(str, Collections.reverseOrder());
        //System.out.println(Arrays.toString(str));
        
        String tmp ="";
        for (int i =0; i< str.length; i++){
            tmp += str[i];
            
        }
        
        return Long.parseLong(tmp) ;
        
        
    }
}


// 1차시도 런타임 에러 발생 
import java.util.*;


class Solution {
    public long solution(long n) {
        long answer = 0;
        String[] str = String.valueOf(n).split("");
        //System.out.println(Arrays.toString(str));       

        Arrays.sort(str, Collections.reverseOrder());
        //System.out.println(Arrays.toString(str));
        
      
        String str2 = String.join("", str); 
        answer = Integer.parseInt(str2); 
        // 정해진 타입인 long이 아니라 Int로 변환해서 런타임에러 발생ㅠㅠ;; 자바 
        
        return answer ;
        
        
    }
}
